DebugLogEnable.h

```diff
@@ -6,12 +6,32 @@
 #undef ASSERT
 #undef ASSERTM
 
+#ifndef LOG_PREAMBLE
+  #define LOG_PREAMBLE LOG_SHORT_FILENAME, __LINE__, __func__
+#endif
+
 #define LOG_SHORT_FILENAME (strrchr(__FILE__, '/') ? strrchr(__FILE__, '/') + 1 : strrchr(__FILE__, '\\') ? strrchr(__FILE__, '\\') + 1 : __FILE__)
-#define LOG_ERROR(...) DebugLog::Manager::get().log(arx::debug::LogLevel::LVL_ERROR, LOG_SHORT_FILENAME, __LINE__, __func__, __VA_ARGS__)
-#define LOG_WARN(...) DebugLog::Manager::get().log(arx::debug::LogLevel::LVL_WARN, LOG_SHORT_FILENAME, __LINE__, __func__, __VA_ARGS__)
-#define LOG_INFO(...) DebugLog::Manager::get().log(arx::debug::LogLevel::LVL_INFO, LOG_SHORT_FILENAME, __LINE__, __func__, __VA_ARGS__)
-#define LOG_DEBUG(...) DebugLog::Manager::get().log(arx::debug::LogLevel::LVL_DEBUG, LOG_SHORT_FILENAME, __LINE__, __func__, __VA_ARGS__)
-#define LOG_TRACE(...) DebugLog::Manager::get().log(arx::debug::LogLevel::LVL_TRACE, LOG_SHORT_FILENAME, __LINE__, __func__, __VA_ARGS__)
+#define LOG_ERROR(...) DebugLog::Manager::get().log(arx::debug::LogLevel::LVL_ERROR, LOG_PREAMBLE, __VA_ARGS__)
+#define  LOG_WARN(...)  DebugLog::Manager::get().log(arx::debug::LogLevel::LVL_WARN, LOG_PREAMBLE, __VA_ARGS__)
+#define  LOG_INFO(...)  DebugLog::Manager::get().log(arx::debug::LogLevel::LVL_INFO, LOG_PREAMBLE, __VA_ARGS__)
+#define LOG_DEBUG(...) DebugLog::Manager::get().log(arx::debug::LogLevel::LVL_DEBUG, LOG_PREAMBLE, __VA_ARGS__)
+#define LOG_TRACE(...) DebugLog::Manager::get().log(arx::debug::LogLevel::LVL_TRACE, LOG_PREAMBLE, __VA_ARGS__)
+
+#if defined(DEBUGLOG_DEFAULT_LOG_LEVEL_ERROR)
+  #define LOG_WARN(...)
+  #define LOG_INFO(...)
+  #define LOG_DEBUG(...)
+  #define LOG_TRACE(...)
+#elif defined(DEBUGLOG_DEFAULT_LOG_LEVEL_WARN)
+  #define LOG_INFO(...)
+  #define LOG_DEBUG(...)
+  #define LOG_TRACE(...)
+#elif defined(DEBUGLOG_DEFAULT_LOG_LEVEL_INFO)
+  #define LOG_DEBUG(...)
+  #define LOG_TRACE(...)
+#elif defined(DEBUGLOG_DEFAULT_LOG_LEVEL_DEBUG)
+  #define LOG_TRACE(...)
+#endif
```

Comment:
This is not necessary for this pr, I think (this p-r is for `LOG_PREAMBLE`)

Reply:
Agreed, not necessary for this PR. This change (along with the earlier L24-28 DebugLog.h) should be in a different PR as they are unrelated to LOG_PREAMBLE